master/buildbot/test/unit/test_schedulers_base.py

```diff
@@ -158,9 +158,10 @@ def gotChange(got_change, got_important):
         d = sched.startConsumingChanges(**kwargs)
 
         def test(_):
-            # check that it registered a callback
-            self.assertEqual(len(self.mq.qrefs), 1)
-            qref = self.mq.qrefs[0]
+            # check that it registered callbacks
+            self.assertEqual(len(self.mq.qrefs), 2)
+
+            qref = self.mq.qrefs[1]
             self.assertEqual(qref.filter, ('changes', None, 'new'))
```

Comment:
So if I defer the enabled state like we're planning, I'm not sure how to wait for the enabled state to change in the test. This is what I was planning on doing to defer the enabled state:

    def _enabledCallback(self, key, msg):
        if msg['enabled']:
            self.enabled = True
            self.activate()
        else:
            d = self.deactivate()
            def fn():
                self.enabled = False
            d.addCallback(fn)

Any suggestions? I'm not quite comfortable with this reactor pattern yet.